.github/workflows/build_and_test.yml

```diff
@@ -127,6 +127,11 @@ jobs:
       - name: Run cargo clippy (release)
         run: cargo hack --feature-powerset clippy --release $CLIPPY_COMMON_ARGS
 
+      - name: Check documentation generation
+        run: cargo doc --all
```

Comment:
Bubbling these up from DM's:

- `--all` => `--workspace` (as I am always confused by `--all` -- though, since we have top level crate without code it probably defaults to that)
- how about `--no-deps`?
- does `--document-private-items` here affect any lints or will we get the same regardless?

Reply:
```suggestion
        run: cargo doc --workspace
```

Reply:
>     * how about `--no-deps`? While this PR does upgrade some deps to get them to be `cargo doc`'d I've just always used `--no-deps` for the less walltime spent; is there nowdays some loss of functionality I am not aware of (assuming we'll publish these in future)?

Good point

>     * would a `--document-private-items` here affect any lints or will we get the same regardless?

I think there won't be any change. I think it's cleaner to pass that instead of passing `-Arustdoc::private_intra_doc_links`.

Reply:
Should we add a `--document-private-items` to this cmdline?

Reply:
I've added it.